gssapi/raw/ext_rfc4178.pyx

```diff
@@ -0,0 +1,54 @@
+GSSAPI="BASE"  # This ensures that a full module is generated by Cython
+
+from gssapi.raw.cython_types cimport *
+from gssapi.raw.cython_converters cimport c_get_mech_oid_set
+from gssapi.raw.creds cimport Creds
+
+from gssapi.raw.misc import GSSError
+
+cdef extern from "python_gssapi_ext.h":
+    OM_uint32 gss_set_neg_mechs(
+        OM_uint32 *minor_status,
+        gss_cred_id_t cred_handle,
+        const gss_OID_set mechs) nogil
+
+
+def set_neg_mechs(Creds cred_handle not None, mechs not None):
+    """
+    set_neg_mechs(Creds cred_handle not None, mechs=None)
+
+    This allows callers to specify the set of security mechanisms that
```

Comment:
We don't typically use the RFC text for descriptions - it tends to be overly lengthy and include a lot of details we don't care about.  See if you can make a shorter description in your own words.